airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -0,0 +1,192 @@
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        replication_start_date = self._replication_start_date
+        if stream_state.get(self.cursor_field):
+            replication_start_date = max(pendulum.parse(stream_state[self.cursor_field]), replication_start_date)
+        params.update(
+            {
+                "items": self.path_param,
+                "since_timestamp": replication_start_date.strftime("%Y-%m-%d %H:%M:%S"),
+            }
+        )
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = super().parse_response(response=response, **kwargs)
+        for record in records:
+            yield record.get(self.data_field)
+
+    def path(self, **kwargs) -> str:
+        return "recents"
+
+
+class Deals(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Deals#getDeals,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+
+class Leads(PaginatedPipedriveStream):
+    """https://developers.pipedrive.com/docs/api/v1/Leads#getLeads"""
+
+
+class Activities(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Activities#getActivities,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+    path_param = "activity"
+
+
+class ActivityFields(PipedriveStream):
+    """https://developers.pipedrive.com/docs/api/v1/ActivityFields#getActivityFields"""
+
+
+class Persons(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Persons#getPersons,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+
+class Pipelines(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Pipelines#getPipelines,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+
+class Stages(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Stages#getStages,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+
+class Users(IncrementalPipedriveStream):
+    """
+    API docs: https://developers.pipedrive.com/docs/api/v1/Users#getUsers,
+    retrieved by https://developers.pipedrive.com/docs/api/v1/Recents#getRecents
+    """
+
+    cursor_field = "modified"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        record_gen = super().parse_response(response=response, **kwargs)
```

Comment:
this override is useless, right?

Reply:
nope, in `users` stream we got values nested in like `user_records = record["data"]["data"]`, so need to yield from `user_records` list. In the other streams `record["data"]["data"]` values are the dicts.